### ISSUE TITLE
'implicit_return' rule config not working

### ISSUE THREAD

**Issue by christopher-hof** [realm/SwiftLint#3029]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
I'm trying to configure the 'implicit_return' so it does not trigger for functions (only for 'getter' and 'closures'). But the following configuration does not seem to work.

```yml
implicit_return:
  excluded:
    - function
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? **0.38.2**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? **Homebrew**
* Paste your configuration file:

```yml
excluded:
  - Carthage
line_length: 140
type_body_length: 500
file_length: 600
type_name:
  max_length:
    warning: 50
identifier_name:
    excluded:
        - id
        - to
opt_in_rules:
    - array_init
    - attributes
    - closure_spacing
    - collection_alignment
    - contains_over_first_not_nil
    - contains_over_range_nil_comparison
    - discouraged_object_literal
    - empty_count
    - empty_string
    - empty_xctest_method
    - enum_case_associated_value_count
    - explicit_init
    - fatal_error_message
    - first_where
    - flatmap_over_map_reduce
    - force_unwrapping
    - function_default_parameter_at_end
    - identical_operands
    - implicit_return
    - joined_default_parameter
    - legacy_random
    - last_where
    - let_var_whitespace
    - literal_expression_end_indentation
    - lower_acl_than_parent
    - modifier_order
    - multiline_arguments
    - multiline_function_chains
    - multiline_parameters
    - operator_usage_whitespace
    - optional_enum_case_matching
    - overridden_super_call
    - override_in_extension
    - pattern_matching_keywords
    - prefer_self_type_over_type_of_self
    - prohibited_super_call
    - reduce_into
    - redundant_nil_coalescing
    - required_enum_case
    - single_test_class
    - sorted_first_last
    - sorted_imports
    - static_operator
    - toggle_bool
    - unavailable_function
    - untyped_error_in_catch
    - unused_control_flow_label
    - unused_declaration
    - unused_import
    - vertical_parameter_alignment
    - yoda_condition

disabled_rules:
    - force_cast

implicit_return:
    excluded:
        - function
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  **No**
* Which Xcode version are you using (check `xcodebuild -version`)? **11.3 (11C29)**

```swift
// This triggers a violation:
private func trimValue(_ value: String?) -> String {
    return (value ?? "").trimmingCharacters(in: CharacterSet.whitespaces)
}
```

**Comment by marcelofabri** [comment]
The rule supports an `included` key (not `excluded`)

**Comment by christopher-hof** [comment]
```
implicit_return:
    included:
        - closure
        - getter
```
also doesn't seem to work. 

```
// This still triggers a violation
private func trimValue(_ value: String?) -> String {
        return (value ?? "").trimmingCharacters(in: CharacterSet.whitespaces)
}
```

**Comment by olvrlrnz** [comment]
I have the same problem and did some more research, but don't really understand the problem.

I checked out Yams and added a new unit test:
```swift
let string = try String(contentsOfFile: "/tmp/swiftlint.yml")

 guard let yaml = try Yams.load(yaml: string) as? [String: Any],
       let ruleConfig = yaml["implicit_return"] as? [String: Any] else {
           throw NSError()
 }

var rule = ImplicitReturnConfiguration()
try rule.apply(configuration: ruleConfig)

XCTAssertTrue(rule.includedKinds.contains(.getter))
XCTAssertTrue(rule.includedKinds.contains(.closure))
XCTAssertFalse(rule.includedKinds.contains(.function))
XCTAssertTrue(rule.severityConfiguration.severity == .warning)
```

Config file:
```yaml
implicit_return:
  severity: warning
  included:
    - getter
    - closure
```

or with default severity:
```yaml
implicit_return:
  included:
    - getter
    - closure
```

the unit test passed for both inputs. However, `swiftlint rules` only works for the first config file, otherwise it outputs
```
Loading configuration from '.swiftlint.yml'
Invalid configuration for 'implicit_return'. Falling back to default.
```

So long story short: if you want to use the `included` directive, you must also use the `severity` directive

**Comment by mattrubin** [comment]
I'm using the configuration:
``` yaml
implicit_return:
  included:
    - closure
```

Most of the time, the rule behaves as expected, but occasionally, I'll get warnings for function and getter returns. I haven't yet been able to figure out if there's some environmental change that's causing some of the lint runs to obey the configuration and some to not, or if it's truly nondeterministic.

I'm not sure if it's relevant, but `swiftlint rules` always shows the configuration as
```
warning, included: [SwiftLintFramework.ImplicitReturnConfiguration.ReturnKind.closure, SwiftLintFramework.ImplicitReturnConfiguration.ReturnKind.function, SwiftLintFramework.ImplicitReturnConfiguration.ReturnKind.getter]
```

**Comment by Igor-Palaguta** [comment]
Looks like after changing `implicit_return` configuration in yaml file we should run `swiftlint --no-cache`

**Comment by StevenMagdy** [comment]
Still doesn't work

**Comment by christopher-hof** [comment]
Release 0.40.0 fixed this issue.
```
implicit_return:
    included:
        - closure
        - getter
```
now works as intended.

**Comment by cprovatas** [comment]
I'm on `0.42.0` and this is not working

**Comment by cprovatas** [comment]
Using a config still doesn't work for me in ` 0.42.0`

**Comment by cprovatas** [comment]
It looks like `ImplicitReturnConfiguration` is initialized without passing any config and just uses the default arguments? 

https://github.com/realm/SwiftLint/blob/5d6e25ae5fb25a24eafa7d2d3d8d0d2efcef46bf/Source/SwiftLintFramework/Rules/Style/ImplicitReturnRule.swift#L5

**Comment by jpsim** [comment]
@cprovatas all rules are initialized with some default configuration, and then that configuration is mutated with the overrides from your configuration file so that's expected.

**Comment by cprovatas** [comment]
I got it working now.  Thanks
